Wrappers/Python/cil/processors/Masker.py

```diff
@@ -0,0 +1,168 @@
+    '''
+
+    @staticmethod
+    def value(mask=None, value=0):
+        r'''This imputes value where mask == 0.
+        :param mask: mask 
+        :type mask: DataContainer
+        :param value: values to be assigned to missing elements
+        :type value: float, default=0
+        '''
+
+        processor = Masker(mode='value', mask=mask, value=value)
+
+        return processor
+    
+    @staticmethod
+    def mean(mask=None, axis=None):
+        r'''This imputes mean where mask == 0. If no axis is specified then operates over flattened array.
+        :param mask: mask 
+        :type mask: DataContainer
+        :param axis: specify axis as int or from 'dimension_labels' to calculate mean. 
+        :type axis: str, int
+        '''
+
+        processor = Masker(mode='mean', mask=mask, axis=axis)
+
+        return processor
+    
+    @staticmethod
+    def median(mask=None, axis=None):
+        r'''This imputes median where mask == 0. If no axis is specified then operates over flattened array.
+        :param mask: mask 
+        :type mask: DataContainer
+        :param axis: specify axis as int or from 'dimension_labels' to calculate median. 
+        :type axis: str, int
+        '''
+
+        processor = Masker(mode='median', mask=mask, axis=axis)
+
+        return processor
+
+    def __init__(self,
+                 mask = None,
+                 mode = 'value',
+                 value = 0,
+                 axis = None ):
+        
+        r'''Processor to fill missing values provided by mask.
+        :param mask: DataContainer containing a boolean array with zeros where outliers are detected
+        :type mask: DataContainer
+        :param mode: a method to fill in missing values (value, mean, median)
+        :type mode: str, default=value
+        :param value: substitute all outliers with a specific value
+        :type value: float, default=0
+        :param axis: specify axis as int or from 'dimension_labels' to calculate mean or median in respective modes 
+        :type axis: str or int
+        :return: DataContainer or it's subclass with masked outliers
+        :rtype: DataContainer or it's subclass   
+        '''
+
+        kwargs = {'mask': mask,
+                  'mode': mode,
+                  'value': value,
+                  'axis': axis}
+
+        super(Masker, self).__init__(**kwargs)
+    
+    def check_input(self, data):
+
+        if self.mask is None:
+            raise ValueError('Please, provide a mask.')
+
+        if not (isinstance(self.mask, DataContainer)):
+            raise TypeError('Mask must be a DataContainer')
+        
+        if not (issubclass(type(data), DataContainer)):
+            raise TypeError('Processor supports only following data types:\n' +
+                            ' - ImageData\n - AcquisitionData\n' +
+                            ' - DataContainer')
+
+        if not (data.shape == self.mask.shape and data.dimension_labels == self.mask.dimension_labels):
```

Comment:
SIRF objects won't have `dimension_labels` so I'd keep the shape check, but move this test to a new test like:

`if hasttr(self.mask, 'dimension_labels') and data.dimension_labels == self.mask.dimension_labels:`